bundle/edu.gemini.pot/src/main/scala/edu/gemini/pot/ModelConverters.scala

```diff
@@ -78,12 +81,17 @@ object ModelConverters {
 
   implicit class SPTarget2SiderealTarget(val sp:SPTarget) extends AnyVal {
 
-    // RCN: this is suuper sketchy but it's what the old code was doing, so ... ?
-    def toNewModel:SiderealTarget =
+    def toSiderealTarget(when: GOption[java.lang.Long]): SiderealTarget =
+       toSiderealTarget(when.asScalaOpt.map(_.longValue))
+
+    def toSiderealTarget(when: Option[Long]): SiderealTarget =
       sp.getTarget.fold(
         too => SiderealTarget.empty.copy(name = too.name),
         sid => sid,
-        ns  => SiderealTarget.empty.copy(name = ns.name, magnitudes = ns.magnitudes, spectralDistribution = ns.spectralDistribution, spatialProfile = ns.spatialProfile)
+        ns  => {
+          val coords = when.flatMap(ns.coords).getOrElse(Coordinates.zero)
+          SiderealTarget.empty.copy(name = ns.name, coordinates = coords, magnitudes = ns.magnitudes, spectralDistribution = ns.spectralDistribution, spatialProfile = ns.spatialProfile)
```

Comment:
At some moment we discussed not to use `empty.copy` in this particular case. The rationale was that if we change `SideralTarget`, e.g adding a new field, this conversion will be forgotten, by using the explicit construct wo should avoid that. What do you think?